### PR TITLE
Syncing flow-state classification tagsets with innovation flows

### PR DESCRIPTION
**Pull request by valentinyanakiev**

### CODE REVIEW

File: src/innovation-flows-sync-tagsets.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          },
+        });
+      } catch (error) {
+        logger.error(error);
+      }
+
+      logger.info(`Synced tagsets for callout with profileId: ${profileID} `);
+    }
+  });
+  await Promise.all(promises);
```
Comment (hero101) [inline src/innovation-flows-sync-tagsets.ts:53]: be aware that if one of the promises rejects or throws, the method will return immediately.

File: src/innovation-flows-sync-tagsets.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  const logUserResult = await alkemioCliClient.logUser();
+  logger.info(`User logged in: ${logUserResult}`);
+
+  const validation = await alkemioCliClient.validateConnection();
+  logger.info(`Connection validated: ${validation}`);
+  await syncAllStates(alkemioCliClient, logger);
+};
+
+const syncJourneyCalloutStates = async (
+  items: any[],
```
Comment (hero101) [inline src/innovation-flows-sync-tagsets.ts:29]: isn't there really any type you can reference?

File: src/innovation-flows-sync-tagsets.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  const logger = createLogger();
+  logger.info('Starting application');
+  try {
+    await syncInnovationFlowClassificationTagsets(logger);
+  } catch (error: any) {
+    logger.error(`Error encountered: ${error.message}`);
+  }
+};
+
+export const syncInnovationFlowClassificationTagsets = async (logger: any) => {
```
Comment (hero101) [inline src/innovation-flows-sync-tagsets.ts:15]: doesn't the logger have any type?

File: src/innovation-flows-sync-tagsets.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  alkemioCliClient: AlkemioCliClient,
+  logger: any
+) => {
+  logger.info('Starting to sync all states');
+  const innovationFlowStates =
+    await alkemioCliClient.sdkClient.innovationFlowStates();
+  const { spaces } = innovationFlowStates.data;
+
+  for (const space of spaces) {
+    if (!space.challenges) continue;
```
Comment (hero101) [inline src/innovation-flows-sync-tagsets.ts:66]: imo the brackets makes the code more readable.

**Review by hero101 — CHANGES_REQUESTED** [review]
I need a call to understand the whole concept of what `syncing` is.

File: src/innovation-flows-sync-tagsets.ts (hunk 1 above)
Comment (valentinyanakiev) [reply src/innovation-flows-sync-tagsets.ts:51]: I am aware, that's why the tricky logic is in try/catch block

**Review by hero101 — APPROVED** [review]

**Pull request merged**